lib/puppet/functions/influxdb/retrieve_token.rb

```diff
@@ -6,30 +6,63 @@
     param 'String', :uri
     param 'String', :token_name
     param 'String', :admin_token_file
+    param 'Boolean', :use_system_store
+    optional_param 'String', :ca_bundle
   end
 
   dispatch :retrieve_token do
     param 'String', :uri
     param 'String', :token_name
     param 'Sensitive', :admin_token
+    param 'Boolean', :use_system_store
+    optional_param 'String', :ca_bundle
   end
 
-  def retrieve_token_file(uri, token_name, admin_token_file)
+  def retrieve_token_file(uri, token_name, admin_token_file, use_system_store, ca_bundle = '')
```

Comment:
Note: I'm far away from being a ruby expert. I always assumed `nil` is a "cleaner" default than empty string. But maybe empty string is the better way.

Reply:
I can't remember exactly what it was, but I had some trouble passing `undef` from Puppet code to this function and this ended up working.  Maybe a default of `nil` here would work with `undef`?